### DISCUSSION TITLE
Integration of DataTables for data display

### DISCUSSION THREAD

**Discussion by StuartFerguson** [hydrostack/hydro#67] [Q&A]
I have just started to rebuild an existing MVC based UI using hydro and so far going pretty well.

One area I am struggling to get my head round though is how to use a library such as DataTables(https://datatables.net/) in my new app.

I have managed to get my data lists displaying in my new app but want to be able to use DataTables to provide all the advanced features such as sorting/paging etc without having to hand roll it.

Are you able to give me a short example on how to convert from a simple table like the samples to using DataTables or any other feature rich JavaScript based grid.

Thanks In Advance,

Stuart

**Comment by kjeske** [comment]
Hi, sorry for late answer. So basically when using the JS data grid, you still will rely on the data from your dedicated endpoint. The sources will be referenced from the [CDN](https://datatables.net/manual/installation#CDN). Then the implementation might look like this:

```c#
// DataTable.cshtml.cs

public class DataTable : HydroComponent
{
    public override void Mount()
    {
        Client.ExecuteJs(
            """
            this.querySelector('[data-table]').DataTable({
                ajax: '/api/myData'
            })
            """);
    }
}
```

```razor
<!-- DataTable.cshtml -->

@model DataTable

<div style="width: 100%">
  <div data-table></div>
</div>
```
